scipy/sparse/csgraph/_flow.pyx

```diff
@@ -29,6 +32,17 @@ class MaximumFlowResult:
         return 'MaximumFlowResult with value of %d' % self.flow_value
 
 
+class MinCostFlowResult:
+
+    def __init__(self, flow_value, residual, flow_cost):
+        self.flow_value = flow_value
```

Comment:
I couldn't guess what `flow_value` was in this context, since you allow general demands (and not just consider a flow from a source to a sink).

Reply:
So, I went through the code; I can see it's just the sum of all flows over all edges. That's a relevant output for things like maximum flow (or minimum cost maximum flow) where you have a fixed source and sink (and indeed it's customary to define the _value of an (s, t)-flow_ accordingly. What I can't see is when that's an object of interest in the case of general demands? If we don't already know of natural use cases, I'd be tempted to shave it off; if one does need it, it's available as `residual.sum()`.

Reply:
And notably, as defined here, `MinCostFlowResult.flow_value` is not the value of an (_s_, _t_)-flow even in the case where only two vertices have supply/demand (as in `MaximumFlowResult`): That's defined to be the sum of the residual over all edges from the source; not all edges in the entire graph.